snuba/clickhouse/formatter/expression.py

```diff
@@ -219,3 +220,31 @@ def _format_datetime_literal(self, exp: Literal) -> str:
 
     def _format_date_literal(self, exp: Literal) -> str:
         return "$D"
+
+    def _anonimize_alias(self, alias: str) -> str:
+        # there may be an alias that looks like `snuba_tags[something]`
+        # the string between the square brackets has the potential to be PII
+        # This function will anonimize that which is between the brackets.
+        # this may erroneously anonimize aliases with square brackets
+        # if they are input by the user, but that is better than leaking PII
+        between_square_brackets_regex = re.compile(r"(?<=\[).*?(?=\])")
+        alias_as_list = list(alias)
+        for match in between_square_brackets_regex.finditer(alias):
+            start_i, end_i = match.span()
+            alias_as_list[start_i:end_i] = "$" + "A" * (end_i - start_i - 1)
```

Comment:
It would be a lot better if this was replaced with a single "$A" instead of an arbitrary number of them. It would make the anonymized queries more comparable.

Reply:
That's an interesting point. If we want comparability, should we do a hash of the tag and put that in instead? That way we would know if it was the same one. Otherwise the comparability would be fake since we'd be saying certain aliases are the same when they aren't

Reply:
That's an interesting idea. However I think what we want to be able to do with the anonymized querylog right now is look at the structure of the query, rather than the specific values of the query. Most of the times when we're optimizing we want to know how a certain column was queried, not the specific values it was queried for.  If we keep to the single `$A` then we can analyze how often particular patterns of query are used.

That's how the rest of the query is anonymized, with generic `$S` for strings etc. I do like the idea of having hashed values though. Maybe we should add a backlog to task to store a "hashed anonymized" version of the query that hashes the values so we can also compare between different values. It might be interesting to see queries that have the same structure, but perform differently with different values.

Reply:
So I tried to do the `$A` solution but it actually made the code look much worse. Essentially when the alias that's substituted into the string is not of the same length, you either have to re-search the string (because it has changed) or but in some logic to re-calculate where the new start/end indexes are. Either way it doesn't look pretty. I implemented the hash version instead because that was actually easier and I think the code looks better. Let me know what you think

Reply:
never mind, I forgot about re.sub

Reply:
ok now it should do exactly what you wanted.